core/src/main/java/edu/wpi/grip/core/Pipeline.java

```diff
@@ -151,20 +249,24 @@ public void onSourceRemoved(SourceRemovedEvent event) {
         }
     }
 
-    public synchronized void addStep(int index, Step step) {
+    public void addStep(int index, Step step) {
```

Comment:
It ensures you lock an unlock correctly. Event if there is an exception.
If there is an exception and you forget to unlock in a `finally` block you could deadlock the entire application.